### PR TITLE
fix: ensure existing labels are retained for secrets in GCP secrets m…

### PR DESCRIPTION
**Pull request by newtondev**
## Problem Statement

When managing existing secrets in GCP Secret Manager using ESO, the original labels on the secret gets removed. This is problematic as the secret here is initially created with KCC (Kubernetes Config Connector) which has the permissions allowed to create secrets. The label `managed-by: external-secrets` is also added so that ESO can manage secret versions.

With ESO we use PushSecret to add secret versions to an existing secret and it only has IAM role `roles/secretmanager.secretVersionManager` which allows it to create secret version and not update the secret at all.

The problem occurs when ESO reconciles and then does not take into account the existing labels already present on the secret and removes these labels causing an update to be triggerred, which causes an error as ESO does not have permissions to update the secret, but only to add secret versions.

This fix ensures that the existing labels are taken into account, thus ensuring there is no diferrerence between the existing secret and the operation that the PushSecret controller wants to invoke, ensuring that only a new secret version gets added.

## Related Issue

Related to #4016 as the use case that was raised by the user rolandkool.

## Proposed Changes

To ensure that the labels of the existing secret are merged with the labels coming from the PushSecret object spec.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by gusfcarvalho** [comment]
hi @newtondev , thanks for your contribution. This change as is will be a breaking change for users, as we are changing behavior of PushSecret Metadata from Replacing content to Merge content. I'd honestly expect `Replace` to still be the defaults, as this is what we have everywhere else even on different API Sets 😄 

IMO - you should add the proper field within the Metadata Struct: 

```
type Metadata struct {
	Annotations map[string]string `json:"annotations"`
	Labels      map[string]string `json:"labels"`
	Topics      []string          `json:"topics,omitempty"`
	MergePolicy EnumOf<Merge|Replace> # Something Like this
}
```
And then add proper flow to control if MergePolicy is Merge or Replace. (making replace the default value, as to not cause breaking changes).

Also, please add some tests 😄

File: pkg/provider/gcp/secretmanager/push_secret.go [hunk 1]
```diff
@@ -79,7 +79,12 @@ func (b *psBuilder) buildMetadata(_, labels map[string]string, _ []*secretmanage
 
 	newLabels := map[string]string{}
 	if metadata.Labels != nil {
-		newLabels = metadata.Labels
+		for key, value := range metadata.Labels {
+			newLabels[key] = value
+		}
```
Comment (Skarlso) [inline pkg/provider/gcp/secretmanager/push_secret.go:84]:
```suggestion
		maps.Copy(newLabels, metadata.Labels)
```

File: pkg/provider/gcp/secretmanager/push_secret.go [hunk 2]
```diff
@@ -79,7 +79,12 @@ func (b *psBuilder) buildMetadata(_, labels map[string]string, _ []*secretmanage
...
+	}
+	for key, value := range labels {
+		newLabels[key] = value
 	}
```
Comment (Skarlso) [inline pkg/provider/gcp/secretmanager/push_secret.go:88]:
```suggestion
	maps.Copy(newLabels, labels)
```

Comment (Skarlso) [inline pkg/provider/gcp/secretmanager/push_secret.go:88]: Replace this entire thing with:

```go
maps.Copy(newLabels, metadata.Labels)
maps.Copy(newLabels, labels)
```

Comment (newtondev) [reply pkg/provider/gcp/secretmanager/push_secret.go:88]: Sure, thanks. Will do.

**Comment by newtondev** [comment]
@gusfcarvalho Thanks for the suggestion, I do however see the kubernetes provider has a mergeTargetMetadata method that seems to do this - https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/kubernetes/metadata.go#L84

**Comment by gusfcarvalho** [comment]
> @gusfcarvalho Thanks for the suggestion, I do however see the kubernetes provider has a mergeTargetMetadata method that seems to do this - https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/kubernetes/metadata.go#L84

This is my point exactly. Instead of introducing a breaking change, we should implement something similar to what kubernetes provider does.

**Comment by newtondev** [comment]
Added unit tests and documentation. Ready for review.

**Comment by newtondev** [comment]
Thank you 🙏

**Pull request merged**